tests/test_edgeql_syntax.py

```diff
@@ -1048,6 +1048,17 @@ def test_edgeql_syntax_name_26(self):
         SELECT ``::Bar;
         """
 
+    @tb.must_fail(errors.EdgeQLSyntaxError, line=2, col=16)
+    def test_edgeql_syntax_name_27(self):
+        """
+        SELECT `$event`;
+        """
+
+    def test_edgeql_syntax_name_28(self):
+        """
+        SELECT `ok$event`;
```

Comment:
I'd ban `$` everywhere, not just when it's the first character. We can relax this ban later, if there's a case for it.

Reply:
As opposed to... literally everything else? Why specifically `$`? There are lots of odd symbols that are allowed inside backticks:
```WITH `[@];-)... "'#//*/` := 42 SELECT `[@];-)... "'#//*/`;```

The simplest usage of backticks is to just allow reserved keywords to potentially be used as identifiers (typically in shapes intended for output), such as `order` or `filter`. However, we also considered that the identifiers might reflect some filenames or even entire paths (for modules, here's a bit of an extreme example: ```WITH MODULE `~/.proj/schema/types` SELECT ...```).

My point is that if you're considering banning `$` everywhere, maybe a bigger review of what we allow in backtick-quoted identifiers is needed.

Reply:
Yeah, I also think banning `$` altogether (as opposed to the beginning of ident) isn't warranted.